lib/spack/spack/config.py

```diff
@@ -234,28 +238,28 @@ def get_section(self, section):
                 if self._raw_data is None:
                     return None
 
-                # support tuples as "or" in the yaml path
-                if isinstance(key, (list, tuple)):
-                    key = first_existing(self._raw_data, key)
-
                 self._raw_data = self._raw_data[key]
 
             for section_key, data in self._raw_data.items():
                 self.sections[section_key] = {section_key: data}
-
         return self.sections.get(section, None)
 
     def write_section(self, section):
-        validate(self.sections, self.schema)
+        full_obj = syaml.syaml_dict()
+        for key, value in self.sections.items():
+            full_obj[key] = value[key]
+        for key in reversed(self.yaml_path):
+            full_obj = syaml.syaml_dict({key: full_obj})
+        validate(full_obj, self.schema)
         try:
             parent = os.path.dirname(self.path)
             mkdirp(parent)
 
-            tmp = os.path.join(parent, '.%s.tmp' % self.path)
+            tmp = os.path.join(parent, '.%s.tmp' % os.path.basename(self.path))
             with open(tmp, 'w') as f:
-                syaml.dump_config(self.sections, stream=f,
+                syaml.dump_config(full_obj, stream=f,
                                   default_flow_style=False)
-            os.path.move(tmp, self.path)
+            os.replace(tmp, self.path)
```

Comment:
I think this will write the config-only portion of the yaml in place of the full yaml for an environment. Does this delete your `specs` list?

Reply:
Surprisingly, no.  It keeps the specs that are installed.  I just tested this to make sure.  I'm not sure this keeps all the top level comments, though.

If you look at the get_section routine, you'll see that all the top level entries are stored in self.sections[].  This includes `spec` and `view`.  When you rebuild the object later, these are both put in their correct locations.